### PR TITLE
Fix Host Function

### PR DESCRIPTION
**Pull request by Szymon20000**
## Description

Fixes: https://github.com/software-mansion/react-native-reanimated/issues/1758
Currently, when we send callback function to UI thread and then back to RN then we will get function wrapper that checks is somebody uses runOnJS. In order to solve the problem, I've added logic that attaches additional info to a function wrapper that allows to get original callback.

## Checklist

- [ ] Included code example that can be used to test this change
- [ ] Updated TS types
- [ ] Added TS types tests
- [ ] Added unit / integration tests
- [ ] Updated documentation
- [ ] Ensured that CI passes

### CODE REVIEW

File: Common/cpp/SharedItems/ShareableValue.cpp [hunk 1]
```diff
@@ -239,8 +250,10 @@ jsi::Value ShareableValue::toJSValue(jsi::Runtime &rt) {
     }
     case ValueType::HostFunctionType: {
       auto hostFunctionWrapper = ValueWrapper::asHostFunctionWrapper(valueContainer);
-      if (hostFunctionWrapper->hostRuntime == &rt) {
+      auto hostRuntime = hostFunctionWrapper->value->hostRuntime;
```
Comment (piaskowyk) [inline Common/cpp/SharedItems/ShareableValue.cpp:253]: I think we can use here `auto &` instead of `auto` to avoid unnecessary copying of objects.

**Review by piaskowyk — APPROVED** [review]

**Comment by enesozturk** [comment]
You made my day 🔥 Looking forward to examples and new release.

**Comment by enesozturk** [comment]
Hi, is this pr included in v2.0.1? Looks like the problem is still continue.

**Comment by enesozturk** [comment]
Hi, I am following issue for my package. It is solved with v2.1.0 actually. When I try it on Expo project with same version (v2.1.0), problem still exist. Not sure if I open an issue in Expo. Wanted to let you know.

**Comment by colinux** [comment]
I found Expo 41 ships reanimated v2.0.1 native modules, but v2.1.0 on JS side
https://github.com/expo/expo/commit/d783eeb1681de085dcf6a433ae7b41b2d238e481 

Not sure why, maybe because the reanimated release was a little late during expo 41 release, so they don't took the risk to ship with it. Anyway I don't think there will be a solution until expo 42 in managed workflow…

**Pull request merged**